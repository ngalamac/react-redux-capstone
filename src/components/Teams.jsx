/* eslint-disable import/no-extraneous-dependencies */
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import NavBar from './NavBar';
import '../styles/Teams.css';

function Teams() {
  const teamsList = useSelector((state) => state.teams.teams);
  const isLoading = useSelector((state) => state.teams.loading);
  const isError = useSelector((state) => state.teams.error);
  const [search, setSearch] = useState('');

  function handleSearch(e) {
    setSearch(e.target.value);
  }

  if (isLoading) {
    return <h1>Loading...</h1>;
  }

  if (isError) {
    return <h1>{isError}</h1>;
  }

  return (
    <>
      <NavBar />
      <div className="search">
        <input type="text" placeholder="Enter a Team Name" value={search} onChange={handleSearch} />
      </div>
      <div className="teams">
        {teamsList
          .filter((team) => team.strTeam.toLowerCase().includes(search.toLowerCase()))
          .map((team) => (
            <Link to={`/team/${team.idTeam}`} key={team.idTeam}>
              <div className="team" key={team.id}>
                <img src={team.strTeamFanart3} alt="Fan" />
                <p>{team.strTeam}</p>
                <span>{team.strTeamShort}</span>
              </div>
            </Link>
          ))}
      </div>
    </>

  );
}

export default Teams;
