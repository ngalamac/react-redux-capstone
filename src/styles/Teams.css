.teams {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  padding: 0 6px;
  gap: 0.5rem;
}

.teams a {
  text-decoration: none;
  color: #fff;
}

.team img {
  width: 185px;
  height: 150px;
  object-fit: cover;
}

.search {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 1rem;
}

.search input {
  outline: 0;
  border: 0;
  width: 350px;
  height: 35px;
  padding: 0 0.5rem;
  font-size: 15px;
  font-weight: bold;
}

.search input[type="text"] {
  color: #000;
}
