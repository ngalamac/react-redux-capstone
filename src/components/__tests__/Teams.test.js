import React from 'react';
import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Teams from '../Teams';

const mockStore = configureMockStore([]);

describe('Teams component works okay', () => {
  test('Teams renders correctly', () => {
    const store = mockStore({
      teams: {
        teams: [],
      },
    });

    const tree = render(
      <Provider store={store}>
        <Teams />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
