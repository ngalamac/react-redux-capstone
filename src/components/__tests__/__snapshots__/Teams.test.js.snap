// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams component works okay Teams renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <img
          alt="premiere league logo"
          src="download.png"
        />
      </header>
      <div
        class="search"
      >
        <input
          placeholder="Enter a Team Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="teams"
      />
    </div>
  </body>,
  "container": <div>
    <header>
      <img
        alt="premiere league logo"
        src="download.png"
      />
    </header>
    <div
      class="search"
    >
      <input
        placeholder="Enter a Team Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="teams"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
